src/pages/RequestCallPage.js

```diff
@@ -203,27 +203,34 @@ class RequestCallPage extends Component {
      *
      * @returns {Object}
      */
-    getFirstAndLastName({login, displayName}) {
-        let firstName;
-        let lastName;
-
-        if (Str.removeSMSDomain(login) === displayName) {
-            firstName = '';
-            lastName = '';
+    getFirstAndLastName({
+        login,
+        displayName,
+        firstName,
+        lastName,
+    }) {
+        let fName;
+        let lName;
+
+        if (firstName && lastName) {
+            fName = firstName;
+            lName = lastName;
+        } else if (Str.removeSMSDomain(login) === displayName) {
+            fName = '';
+            lName = '';
```

Comment:
Hmm, Please confirm this on Slack first. and use early return.

Reply:
Sorry, I don't understand. Do you mean whether we need to fallback to displayName? I don't think we need to even fallback to displayName. Are there any cases where we won't have firstName and lastName.

Reply:
Yes, you have already added the condition check for such a case. If user is using email or phone but haven't set the name yet

Reply:
So, this should be okay?
```
getFirstAndLastName({
    firstName,
    lastName,
}) {
    if (firstName && lastName) {
        return {firstName, lastName};
    }
        
    return {firstName: '', lastName: ''};
}
```
Any suggestion?

Reply:
Oh, I was missing a case where user can either have only firstName or lastName set. 
```
    getFirstAndLastName({
        firstName,
        lastName,
    }) {
        if (firstName && lastName) {
            return {firstName, lastName};
        }
        if (firstName) {
            return {firstName, lastName: ''};
        }
        if (lastName) {
            return {firstName: '', lastName};
        }
        return {firstName: '', lastName: ''};
    }
```
This should be okay.. Any suggestion,?

Reply:
only this is fine.

if (firstName || lastName) {
            return {firstName: firstName || ' ' , lastName: lastname || ' ' };
        }

Reply:
Well. In that case we can completely get rid of `getFirstAndLastName()` function. This changes in state initialization should be enough.
```
        this.state = {
            firstName: props.myPersonalDetails.firstName || '',
            hasFirstNameError: false,
            lastName: props.myPersonalDetails.lastName || '',
```
Any thoughts?

Reply:
I meant to say use first and last name as this block and use the existing code to fallback to displayname. Also, fix the displayName code so that there is no space issue.

Reply:
So, our final change will look like this?
```
    getFirstAndLastName({
        login,
        displayName,
        firstName,
        lastName,
    }) {
        if (firstName || lastName) {
            return {firstName: firstName || '', lastName: lastName || ''};
        }
        if (Str.removeSMSDomain(login) === displayName) {
            return {firstName: '', lastName: ''};
        }

        let fName = '';
        let lName = '';
        const firstSpaceIndex = displayName.indexOf(' ');
        const lastSpaceIndex = displayName.lastIndexOf(' ');
        if (firstSpaceIndex === -1) {
            fName = displayName;
            lName = '';
        } else {
            fName = displayName.substring(0, firstSpaceIndex).trim();
            lName = displayName.substring(lastSpaceIndex).trim();
        }
        return {firstName: fName, lastName: lName};
    }
```

Reply:
Lets leave this  
```
 if (Str.removeSMSDomain(login) === displayName) {
            return {firstName: '', lastName: ''};
        }
 ```

Reply:
If both firstName and lastName are not set by the user, the above block of code will be executed. So we will be needing that block of code.

Reply:
Instead of 
```
    if (firstName || lastName) {
         return {firstName: firstName || '', lastName: lastName || ''};
    }
```
if we use following block of code, the block you mentioned can be removed.
```
        if ( typeof firstName !== undefined || typeof lastName !== undefined) {
            return {firstName: firstName || '', lastName: lastName || ''};
        }
```

Reply:
Thanks. Will update accordingly.